v1/src/main/java/com/google/cloud/teleport/datadog/DatadogEventWriter.java

```diff
@@ -63,6 +62,7 @@ public abstract class DatadogEventWriter
   private static final Integer MAX_BATCH_COUNT = 1000;
   private static final Logger LOG = LoggerFactory.getLogger(DatadogEventWriter.class);
   private static final long DEFAULT_FLUSH_DELAY = 2;
+  private static final Long MAX_BUFFER_SIZE = 5L * 1000 * 1000; // 5MB
```

Comment:
Is the limit actually 5mb? or a bit higher? It only considers the payload? Maybe we could have the limit a bit lower here in case there's anything else considered (a really unlikely example would be the `[` `]` as you are sending the array), but my concern is if for example, HTTP headers are taken into account.